pkgs/shells/fish/default.nix

```diff
@@ -13,9 +13,11 @@
 , pcre2
 , gettext
 , ncurses
-, python3
 , cmake
 
+, usePython ? true # used for autocompletion generation from manpages and browser config editing
+, python3 ? null
```

Comment:
Would this prevent adding `python3.pkgs.sphinx` to `nativeBuildInputs`? I was planning on (sometime in the future) adding that in order to be able to `fetchFromGitHub` and build the docs ourselves, rather than relying on them being present in the published tarball.

Reply:
I think `python37Packages.sphinx` may be the appropriate thing to use then?

Reply:
Only reason I opted for `python3.pkgs` is because it's already there; they appear to be the same thing:

```
pkgs/top-level/all-packages.nix
9525:  python3Packages = python3.pkgs;
```

To clarify: your suggestion would be to add `python3Packages` to the top-level args? Or maybe add `sphinx` and modify the `callPackage` to have `inherit (python3Packages) sphinx;`...

Reply:
I think adding `python3packages` to the top-level args would be most appropriate. While that means instead of having to provide `python3` one has to provide `python3Packages`, which makes the improvement of not having to worry about what Python to pass kind of moot, calling derivations with only the packages they need instead of all of them is really an orthogonal problem - lots of other packages require the whole of `python3Packages` as well.
Ideally we would develop a pattern to fix that and apply it treewide (maybe it should be the `all-packages.nix` modification you're suggesting).

Reply:
infinisil brings up a good point. One option is that when you make the PR you remove the `? null` and group `python3` with the other arguments, but leave the rest in place, and use `python3.pkgs.sphinx`. The other is just to ignore that, since one is a build time dependency on Python and the other is a run time dependency on Python so it is not crucial to keep them the same. It may even be advantageous for them not to be the same if one wants to use a different interpreter for each.

Reply:
I'd say I leave this as is and you do whatever you eventually decide is the best solution when you make your PR. :)

Reply:
I concur. I'll do some more brainstorming, and see if I can't come up with any better solutions in the meantime. Thanks for the information/bikeshedding :)